dask_sql/physical/rex/core/literal.py

```diff
@@ -167,18 +168,19 @@ def convert(
             "TimestampNanosecond",
         }:
             unit_mapping = {
-                "Second": "s",
-                "Millisecond": "ms",
-                "Microsecond": "us",
-                "Nanosecond": "ns",
+                "TimestampSecond": "s",
+                "TimestampMillisecond": "ms",
+                "TimestampMicrosecond": "us",
+                "TimestampNanosecond": "ns",
             }
+            numpy_unit = unit_mapping.get(literal_type)
             literal_value, timezone = rex.getTimestampValue()
             if timezone and timezone != "UTC":
                 raise ValueError("Non UTC timezones not supported")
+            elif timezone is None:
+                literal_value = datetime.fromtimestamp(literal_value // 10**9)
```

Comment:
If gpuCI still fails, you can try adding `literal_value = str(literal_value)` after this line (but still in the `elif` block).

Reply:
> If gpuCI still fails, you can try adding `literal_value = str(literal_value)` after this line (but still in the `elif` block).

Thanks Sarah. I tried that but this still fails. Here is more info on the failure:

```
self = <dask_sql.physical.rex.core.call.ReduceOperation object at 0x7f4717a097c0>
operands = ('2001-03-09', datetime.timedelta(days=90)), kwargs = {}

    def reduce(self, *operands, **kwargs):
        if len(operands) > 1:
            if any(
                map(
                    lambda op: is_frame(op) & pd.api.types.is_datetime64_dtype(op),
                    operands,
                )
            ):
                operands = tuple(map(as_timelike, operands))
>           return reduce(partial(self.operation, **kwargs), operands)
E           TypeError: can only concatenate str (not "datetime.timedelta") to str

dask_sql/physical/rex/core/call.py:137: TypeError
```

Perhaps `def reduce` or `def as_timelike` need updating to support the interval type that the Rust code is returning?

It looks like @charlesbluca may be familiar with these methods and maybe has some insight here.

Reply:
Curious if this change is still needed

Reply:
I don't think we need the string casting, but it seems like the `datetime.fromtimestamp` is necessary to get the proper timezone for the datetime